subprojects/workers/src/main/java/org/gradle/workers/internal/ClassLoaderStructure.java

```diff
@@ -19,7 +19,9 @@
 import com.google.common.base.Objects;
 import org.gradle.internal.classloader.ClassLoaderSpec;
 
-public class ClassLoaderStructure {
+import java.io.Serializable;
+
+public class ClassLoaderStructure implements Serializable {
```

Comment:
I've added serializers for this.  Unfortunately, since `WorkerProtocol` is not the protocol that we use at the lowest level (i.e. we use `RequestProtocol` to tunnel `WorkerProtocol` over to the worker) we really need to serialize `RequestProtocol` and have a way to add a serializer to use for a specific implementation class.  In other words, the type parameter on the arguments of `RequestProtocol` is `Object[]` so `MethodArgsSerializer` thinks it should always use java serialization for that argument.  What I did was implement a serializer for `RequestProtocol` and added a way to register a serializer on the `WorkerProcessBuilder` for the arguments of whatever protocol is being implemented with that builder.  This works, but we could probably simplify this some more.  I'm planning on splitting the serializer stuff off as a separate PR so we can clean that up separately.